src/main/flight/pid.c

```diff
@@ -766,7 +789,8 @@ void pidController(const pidProfile_t *pidProfile, const rollAndPitchTrims_t *an
         pidData[axis].P = pidData[axis].P * getThrottlePAttenuation() * setPointPAttenuation[axis];
         pidData[axis].I = temporaryIterm[axis] * getThrottleIAttenuation() * setPointIAttenuation[axis]; // you can't use pidData[axis].I to calculate iterm or with tpa you get issues
         pidData[axis].D = pidData[axis].D * getThrottleDAttenuation() * setPointDAttenuation[axis];
-        const float pidSum = pidData[axis].P + pidData[axis].I + pidData[axis].D;
+        float pidSum = pidData[axis].P + pidData[axis].I + pidData[axis].D;
+        pidSum = pidSum * scaledAxisPid[axis];
```

Comment:
I'm trying to refactor this method to make it really easy to understand. Also there are a couple of things that can be optimized, e.g. the calculation of errorBoostAxis and errorLimitAxis (some operations are executed over and over for each pid loop and axis) and the if-else branch used for the nfe mode.

Can we rename scaledAxisPid -> axisLockMultiplier or something like that?

Reply:
Error boost needs to be calculated per axis as the effect depends on how much error each individual axis has. Same as the limit.  Yes this can be named better, but the things you mentioned canr be changed to much except nfe mode could be made into a case statement.

Reply:
Of course I see the logic, I was talking of some multiplication that can be avoided to be repeated over and over simply using a couple of global variables. I'll show you how later. I was asking if you agree with the name I'm suggesting. I think that renaming "scaledAxisPid" into "axisLockMultiplier" could help to better understand what's the meaning of that variable without having to look at how it's used.

Reply:
axisLockScale is probably a better name. axisLockMultiplier is already used to determine how strong of an effect this has, and i think i know what you mean by adding global variables, I think thats how i implemented things into 1.0.0, I mean I don't want to put a ton more work into maintaining this code base, this is kind of just a last hurra for the butterflight codebase.
I hope that makes sense.

Reply:
Yes absolutely makes sense!